src/client.rs

```diff
@@ -46,16 +46,26 @@ impl Client {
     /// (like unreachable host) so it's important that you handle those
     /// errors.
     pub fn get_connection(&self) -> RedisResult<Connection> {
-        connect(&self.connection_info, None)
+        connect(&self.connection_info, None, None, None)
     }
 
     /// Instructs the client to actually connect to redis with specified
     /// timeout and returns a connection object.  The connection object
     /// can be used to send commands to the server.  This can fail with
     /// a variety of errors (like unreachable host) so it's important
     /// that you handle those errors.
-    pub fn get_connection_with_timeout(&self, timeout: Duration) -> RedisResult<Connection> {
-        connect(&self.connection_info, Some(timeout))
+    pub fn get_connection_with_timeouts(
```

Comment:
I am picking this up again, but I am struggling how I can provide a stream to the `Connection`. Can give me a few pointers in the right direction?

I could imagine having something like `Connection::from_tcp_stream()` which would create a `redis::Connection` struct for me. But as far as I can tell until now I am forced to go through `redis::Client` to get a connection, which makes a lot of sense.

The documentation says, that `redis::Client` might develop into a connection pool. My intuition would say, that a connection pool needs all the "settings" of the connections, to have a few of them completely ready. Or I guess `get_connection_with_options()` might also be good.

In fact, I think the current `get_connection_with_timeout()` would imply, that I provide a timeout until I want to have a connection ready to use. But this is not the case now and I think not that easy to provide.

Reply:
THe async client provides a way to directly construct a `Connection` without going through `Client` so you should be able to have a similar API https://github.com/mitsuhiko/redis-rs/blob/d4dfd43a0f8ea87319fc5f064fae6930550c4c4e/src/aio.rs#L283

Reply:
Still struggling:

If I wanted to make it analogous to the `aio::Connection`, I would make `connection::Connection` generic over a parameter that is `Read + Write`. That is however not possible, because then I cannot implement `set_write_timeout` and `set_read_timeout`.

Otherwise I could make different construction methods like: `Connection::from_tcp_stream()`, `Connection::from_unix_stream()` and `Connection::from_tls_stream()`

Reply:
You can do `impl Connection<ActualConnection> { fn set_read_timeout ... }` to only support those methods on an underlying connection that supports it. (`ActualConnection` should probably have a better name, but ...)